Client/Frontend/Browser/HomePanel/FavoritesViewController.swift

```diff
@@ -275,6 +282,13 @@ class FavoritesViewController: UIViewController, Themeable {
         }
     }
     
+    @objc func handleTapGesture(gesture: UITapGestureRecognizer) {
+        // Tap gesture only actionable with sponsored images.
+        if backgroundImage.info?.isSponsored == true && gesture.state == .ended {
```

Comment:
I would reverse the condition and put `gesture.state == .ended` first, so we don't check for sponsored data when not needed.